api/lib/validate.js

```diff
@@ -1,99 +1,137 @@
 'use strict';
 var _ = require('lodash');
-var errors = require('./errors.js');
+var InvalidRequestError = require('./errors.js').InvalidRequestError;
 var validator = require('./schema-validator');
 var ripple = require('ripple-lib');
 var utils = require('./utils');
 
-function addOptionalOption(validatorFunction) {
-  return function(parameter, optional) {
-    if (optional && parameter === undefined) {
-      return null;
-    }
-    return validatorFunction(parameter);
-  };
+function isValidBoolString(boolString) {
+  return (boolString === 'true' || boolString === false);
+}
+
+function error(text) {
+  return new InvalidRequestError(text);
+}
+
+function invalid(type, value) {
+  return error('Not a valid ' + type + ': ' + value.toString());
+}
+
+function missing(name) {
+  return error('Missing parameter: ' + name);
+}
+
+function validateAddress(address) {
+  if (!ripple.UInt160.is_valid(address)) {
+    throw error('Parameter is not a valid Ripple address: account');
+    // thow invalid('Ripple address', address);
+  }
 }
 
-function validateAccount(account) {
-  if (!ripple.UInt160.is_valid(account)) {
-    return new errors.InvalidRequestError(
-      'Parameter is not a valid Ripple address: account');
+function validateAddressAndSecret(obj) {
+  var address = obj.address;
+  var secret = obj.secret;
+  if (!address) {
+    throw missing('address');
+  }
+  if (!secret) {
+    throw missing('secret');
+  }
+  if (secret[0] !== 's' || !ripple.UInt160.is_valid('r' + secret.slice(1))) {
+    throw invalid('Ripple secret', secret);
+  }
+  if (!ripple.Seed.from_json(secret).get_key(address)) {
+    throw error('Secret does not match specified address', secret);
   }
-  return null;
 }
 
 function validateCurrency(currency) {
   if (!validator.isValid(currency, 'Currency')) {
-    return new errors.InvalidRequestError(
-      'Parameter is not a valid currency: currency');
+    throw error('Parameter is not a valid currency: currency');
```

Comment:
No, the commented ones are TODOs. I don't want to change all the test cases yet.